realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/MongoCollection.java

```diff
@@ -212,18 +234,18 @@ public MongoNamespace getNamespace() {
      *
      * @return the find iterable interface
      */
-    // FIXME: fix find implementation - ignore this code for code review
-    public Task<Iterable<DocumentT>> find() {
-        return dispatcher.dispatchTask(() ->
-                osMongoCollection.find()
-        );
+    public FindIterable<DocumentT> find() {
+        return new FindIterable<>(dispatcher, osMongoCollection.find());
     }
 
-    // FIXME: fix find implementation - ignore this code for code review
-    public Task<Iterable<DocumentT>> find(final FindOptions options) {
-        return dispatcher.dispatchTask(() ->
-                osMongoCollection.find(options)
-        );
+    /**
+     * Finds all documents in the collection that match the given filter.
+     *
+     * @param options a {@link FindOptions} struct
+     * @return the find iterable interface
```

Comment:
Not sure what this means?

Reply:
It is the wrong description, it should specify that it uses options to build the query.